server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java

```diff
@@ -230,12 +234,20 @@ public static void readMultiLineFormat(BytesReference data,
                             searchRequest.routing(nodeStringValue(value, null));
                         } else if ("allow_partial_search_results".equals(entry.getKey())) {
                             searchRequest.allowPartialSearchResults(nodeBooleanValue(value, null));
+                        } else if ("expand_wildcards".equals(entry.getKey()) || "expandWildcards".equals(entry.getKey())) {
+                            expandWildcards = value;
+                        } else if ("ignore_unavailable".equals(entry.getKey()) || "ignoreUnavailable".equals(entry.getKey())) {
+                            ignoreUnavailable = value;
+                        } else if ("allow_no_indices".equals(entry.getKey()) || "allowNoIndices".equals(entry.getKey())) {
+                            allowNoIndices = value;
+                        } else if ("ignore_throttled".equals(entry.getKey()) || "ignoreThrottled".equals(entry.getKey())) {
+                            ignoreThrottled = value;
```

Comment:
Maybe we should rather improve IndicesOptions#fromMap and make sure that we don't ignore what we don't recognize? I believe we have the same problem in create and restore snapshot API. Could we make the method remove what it parses and check that the map is left empty after all the parsing? Just an idea.

Reply:
this is unrelated. In this case we need to know if the key is valid and fromMap doesn't tell me what it consumed.

Reply:
that is why I am proposing to change/improve fromMap and to make other callers of it stricter as well. It does not have to be as part of this PR but I think it would be a good change.

Reply:
not sure I understand what you mean. can you make an example?

Reply:
I haven't tried this so it may not work but I am wondering if fromMap removes what it parses from the map provided as argument, and we also update our current "parsing" code to remove the corresponding entry from the map whenever a field is parsed, we can enforce that the map needs to be empty at the end (hence making parsing strict) without having to duplicate the parsing code. Does that make sense?